test/apps/iodemo/io_demo.cc

```diff
@@ -281,6 +296,12 @@ class P2pDemoCommon : public UcxContext {
 
     private:
         void*                     _buffer;
+        // With AM API need to send a header with every iov element, because:
+        // 1. IOVs (chunks) may arrive OOO, therefore receiver needs to know
+        // total message size to initialize receive IOV buffer properly.
```

Comment:
i think AM headers always arrive in order, no?

Reply:
headers are bundled with data and we can get OOO in case of eager multirail

Reply:
so am callbacks are actually out of order?
i think this is not very intuitive, or at least some users would like strict order, like tags. is it documented in UCP API?